src/Tickets/Admin/All_Tickets/List_Table.php

```diff
@@ -498,31 +567,111 @@ public function modify_sort_args( $args ): array {
 
 		return $args;
 	}
 
 	/**
-	 * Get the query args for the list table.
+	 * Modify the filter arguments.
 	 *
 	 * @since TBD
 	 *
+	 * @param array $args The arguments used to query the tickets for the All Tickets Table.
+	 *
 	 * @return array
 	 */
-	public function get_query_args(): array {
+	public function modify_filter_args( $args ) {
+		$filter = tribe_get_request_var( self::$status_key, self::$default_status );
+
+		if ( 'all' === $filter ) {
+			return $args;
+		}
+
+		switch ( $filter ) {
+			case 'active':
+				$args['meta_query'][] = [
+					'key'     => '_ticket_start_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '<=',
+					'type'    => 'DATETIME',
+				];
+				$args['meta_query'][] = [
+					'key'     => '_ticket_end_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '>',
+					'type'    => 'DATETIME',
+				];
+				break;
+			case 'past':
+				$args['meta_query'][] = [
+					'key'     => '_ticket_end_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '<=',
+					'type'    => 'DATETIME',
+				];
+				break;
+			case 'upcoming':
+				$args['meta_query'][] = [
+					'key'     => '_ticket_start_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '>',
+					'type'    => 'DATETIME',
+				];
+				break;
+			case 'discounted':
+				$args['meta_query'][] = [
+					'key'     => '_sale_price',
+					'compare' => 'EXISTS',
+				];
+				$args['meta_query'][] = [
+					'key'     => '_sale_price_start_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '<',
+					'type'    => 'DATETIME',
+				];
+				$args['meta_query'][] = [
+					'key'     => '_sale_price_end_date',
+					'value'   => current_time( 'mysql' ),
+					'compare' => '>',
+					'type'    => 'DATETIME',
+				];
+				break;
+		}
+
+		return $args;
+	}
+
+	/**
+	 * Get the query args for the list table.
+	 *
+	 * @since TBD
+	 *
+	 * @return array|bool
+	 */
+	public function get_query_args() {
 		$current_page = $this->get_pagenum();
 		$per_page     = $this->get_items_per_page( $this->per_page_option );
 
+		$provider_options = $this->get_provider_options();
+		$default_provider = empty( $provider_options ) ? '' : key( $provider_options );
+		$current_provider = tribe_get_request_var( self::$provider_key, $default_provider );
+
 		$args = [
-			'offset'             => ( $current_page - 1 ) * $per_page,
-			'posts_per_page'     => $per_page,
-			'return_total_found' => true,
+			'all_tickets_list_table' => true,
+			'offset'                 => ( $current_page - 1 ) * $per_page,
+			'posts_per_page'         => $per_page,
+			'return_total_found'     => true,
+			'post_type'              => $current_provider,
 		];
 
+		$args = $this->modify_filter_args( $args );
 		$args = $this->modify_sort_args( $args );
 
+		if ( isset( $args['meta_query'] ) && count( $args['meta_query'] ) > 1 ) {
+			$args['meta_query']['relation'] = 'AND';
+		}
```

Comment:
I would do this inside the method i am modifying the `meta_query`. Just so i am sure, i didnt modify the `relation` if i didn't modify the `meta_query`